storage/client/queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of the {@link QueueClient QueueClients}
+ * and {@link QueueAsyncClient QueueAsyncClients}, calling {@link QueueClientBuilder#buildClient() buildClient} constructs an
+ * instance of QueueClient and calling {@link QueueClientBuilder#buildAsyncClient() buildAsyncClient} constructs an instance of
+ * QueueAsyncClient.
+ *
+ * <p>The client needs the endpoint of the Azure Storage Queue service, name of the queue, and authorization credentials.
+ * {@link QueueClientBuilder#endpoint(String) endpoint} gives the builder the endpoint and may give the builder the
+ * {@link QueueClientBuilder#queueName(String) queueName} and a {@link SASTokenCredential} that authorizes the client.</p>
+ *
+ * <p><strong>Instantiating a synchronous Queue Client with SAS token</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueClient.instantiation.sastoken}
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client with SAS token</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueAsyncClient.instantiation.sastoken}
+ *
+ * <p>If the {@code endpoint} doesn't contain the queue name or {@code SASTokenCredential} they may be set using
+ * {@link QueueClientBuilder#queueName(String) queueName} and {@link QueueClientBuilder#credential(SASTokenCredential) credential}.</p>
+ *
+ * <p><strong>Instantiating a synchronous Queue Client with credential</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueClient.instantiation.credential}
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client with credential</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueAsyncClient.instantiation.credential}
+ *
+ * <p>Another way to authenticate the client is using a {@link SharedKeyCredential}. To create a SharedKeyCredential
+ * a connection string from the Storage Queue service must be used. Set the SharedKeyCredential with
+ * {@link QueueClientBuilder#connectionString(String) connectionString}. If the builder has both a SASTokenCredential and
+ * SharedKeyCredential the SharedKeyCredential will be preferred when authorizing requests sent to the service.</p>
+ *
+ * <p><strong>Instantiating a synchronous Queue Client with connection string.</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueClient.instantiation.connectionstring}
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client with connection string.</strong></p>
+ * {@codesnippet com.azure.storage.queue.queueAsyncClient.instantiation.connectionstring}
+ *
+ * @see QueueClient
+ * @see QueueAsyncClient
+ * @see SASTokenCredential
+ * @see SharedKeyCredential
+ */
+public final class QueueClientBuilder {
+    private static final String ACCOUNT_NAME = "accountname";
+    private final List<HttpPipelinePolicy> policies;
+
+    private URL endpoint;
+    private String queueName;
+    private SASTokenCredential sasTokenCredential;
+    private SharedKeyCredential sharedKeyCredential;
```

Comment:
> An early pass on a few of the files in this PR. It would be wise to get this codebase updated to the latest Azure Core version. What is the plan there?

Queue, file will have the same schedule with Blob to move to core preview 3. @alzimmermsft suggested to put it to storage preview 3. @alzimmermsft please correct me if you have anything else on top of your plan.

Reply:
What was the tokenCredential used for in queue?